---
title: General-purpose GPU hashing data structures and their application in accelerated genomics
date: 2026-02-19
type: paper
bibtex: |
  @article{JUNGER2022256,
  title = {General-purpose GPU hashing data structures and their application in accelerated genomics},
  journal = {Journal of Parallel and Distributed Computing},
  volume = {163},
  pages = {256-268},
  year = {2022},
  issn = {0743-7315},
  doi = {https://doi.org/10.1016/j.jpdc.2022.01.006},
  url = {https://www.sciencedirect.com/science/article/pii/S0743731522000132},
  author = {Daniel Jünger and Robin Kobus and André Müller and Christian Hundt and Kai Xu and Weiguo Liu and Bertil Schmidt},
  keywords = {GPUs, Hash tables, Bioinformatics},
  abstract = {A broad variety of applications relies on associative data structures that exclusively support insert, retrieve, and delete operations. Hash maps represent such a class of effective dictionary implementations. Properties such as amortized constant time complexity for these table operations as well as a compact memory layout make them versatile data structures with manifold applications in data analytics and artificial intelligence. The rapidly growing amount of data emerging in many scientific fields can often only be tackled with modern massively parallel accelerators such as GPUs. Numerous GPU hash table implementations have been proposed over the recent years. However, most of these implementations lack flexibility in order to be used in existing analytics pipelines or suffer from significant performance degradation for certain application scenarios. As a more recent approach, the WarpCore framework aims to alleviate these aforementioned restrictions by placing a focus on both versatility and performance. In this work we reflect the key concepts of the WarpCore library and provide an extensive performance evaluation against the state-of-the-art. We further explore how WarpCore can be used for accelerating two bioinformatics applications (metagenomic classification and k-mer counting) with significant speedups.}
  }
doi: https://doi.org/10.1016/j.jpdc.2022.01.006
---

## Summary

## Key Contributions

## Notes

